File: content/problems/0413-arithmetic-slices/solutions/ending-slice-dp/solution.cpp
class Solution {
    public: int numberOfArithmeticSlices(vector<int>& nums){
        int current=0,answer=0;
        for(int i=2;i<(int)nums.size();++i){
            if((long long)nums[i]-nums[i-1]==(long long)nums[i-1]-nums[i-2])++current;
            else current=0;
            answer+=current;
        }
        return answer;
    }
};

File: content/problems/0413-arithmetic-slices/solutions/ending-slice-dp/Solution.kt
class Solution {
    fun numberOfArithmeticSlices(nums:IntArray):Int{
        var current=0;
        var answer=0;
        for(i in 2 until nums.size){
            current=if(nums[i].toLong()-nums[i-1]==nums[i-1].toLong()-nums[i-2])current+1 else 0;
            answer+=current
        };
        return answer
    }
}
